File: system-test/java/src/main/java/com/optivem/shop/dsl/core/usecase/external/clock/usecases/base/BaseClockUseCase.java
package com.optivem.shop.dsl.core.usecase.external.clock.usecases.base;

import com.optivem.shop.dsl.driver.port.external.clock.ClockDriver;
import com.optivem.shop.dsl.core.shared.BaseUseCase;
import com.optivem.shop.dsl.core.shared.UseCaseContext;

public abstract class BaseClockUseCase<TSuccessResponse, TSuccessVerification> extends BaseUseCase<ClockDriver, TSuccessResponse, TSuccessVerification> {
    protected BaseClockUseCase(ClockDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}

File: system-test/dotnet/Dsl.Core/UseCase/External/Clock/UseCases/Base/BaseClockUseCase.cs
using Driver.Port.External.Clock;
using Driver.Port.External.Clock.Dtos;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Clock.UseCases.Base;

public abstract class BaseClockUseCase<TResponse, TVerification>
    where TVerification : ResponseVerification<TResponse>
{
    protected readonly IClockDriver _driver;
    protected readonly UseCaseContext _context;

    protected BaseClockUseCase(IClockDriver driver, UseCaseContext context)
    {
        _driver = driver;
        _context = context;
    }

    public abstract Task<ClockUseCaseResult<TResponse, TVerification>> Execute();
}



